pkg/observer/data.go

```diff
@@ -54,14 +54,19 @@ func add(r *bytes.Reader, m *dataapi.MsgData) error {
 	return DataAdd(m.Id, msgData)
 }
 
-func DataGet(id dataapi.DataEventId) ([]byte, error) {
-	data, ok := dataMap.Get(id)
+func DataGet(desc dataapi.DataEventDesc) ([]byte, error) {
+	data, ok := dataMap.Get(desc.Id)
 	if !ok {
-		return nil, fmt.Errorf("failed to find data for id: %v", id)
+		return nil, fmt.Errorf("failed to find data for id: %v", desc.Id)
+	}
+
+	// make sure we did not loose anything on the way through ring buffer
+	if len(data) != int(desc.Size-desc.Leftover) {
+		return nil, fmt.Errorf("failed to get correct data for id: %v", desc.Id)
```

Comment:
Should we remove the id from the dataMap before we return here?

Reply:
yes, I moved it up in the new push, thanks